shlomobot_pytest/utils.py

```diff
@@ -77,18 +89,58 @@ def calculate_total_deducted_score(
     return min(total_points_deducted, max_points_deducted)
 
 
-def import_pyfile(module_name: str) -> ModuleType:
+def import_pyfile(py_filename: str) -> ModuleType:
     """Extracts the module from a given filename"""
-    stripped_module_name = module_name.removesuffix(".py")
+    stripped_module_name = py_filename.removesuffix(".py")
 
     return import_module(stripped_module_name)
 
 
-def get_functions_from_files(file_list: list[str]) -> list[FunctionType]:
+def get_functions_from_files(file_list: list[str]) -> Generator:
     """
     Extracts all functions from the files in file_list
     Creates a generator that returns a filename, FunctionType object pair
     """
-    for filename in file_list:
-        module = import_pyfile(filename)
-        yield filename, extract_functions(module)
+    for py_filename in file_list:
+        module = import_pyfile(py_filename)
+        yield py_filename, extract_functions(module)
+
+
+def get_clean_function_lines(function: FunctionType) -> list[str]:
+    """Counts the amount on non comment or docstring lines in a function code"""
+
+    clean_lines = []
+    function_code = inspect.getsource(function)
+    docstring_state = {"is_active": False, "value": ""}
+
+    for line in function_code.splitlines():
+        if not re.search(COMMENT_REGEX, line) and not re.search(
+            ONE_LINE_DOCSTRING_REGEX, line
+        ):
+            open_match = re.match(OPEN_DOCSTRING_REGEX, line)
+            close_match = re.match(CLOSE_DOCSTRING_REGEX, line)
+
+            if open_match and not docstring_state["is_active"]:
+                docstring_state["is_active"] = True
+                docstring_state["value"] = open_match.group()
+            elif (
+                close_match
+                and docstring_state["is_active"]
+                and (close_match.group() == docstring_state["value"])
+            ):
+                docstring_state["is_active"] = False
+                docstring_state["value"] = ""
+            elif not docstring_state["is_active"]:
+                clean_lines.append(line)
+
+    return clean_lines
+
+
+def function_contains_regex(regex: str, function: FunctionType) -> bool:
```

Comment:
regex might also be a compiled regex

```suggestion
def function_contains_regex(regex: str | re.Pattern, function: FunctionType) -> bool:
```